### PR TITLE
Feature/new usdc swap

### PR DESCRIPTION
**Pull request by andropixels**
@Elaixzar @saadjhk 
please review the changes and merge 

*major chnages here 
1-removed wallets
2-removed poller 
3-evm tokens type chnaged to BridgeToken
4-added errors 
5-removed unneccessary console log statement 
6-added neccessary exports.

### CODE REVIEW

File: src/lib/chains/evm/evmErrors.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+export enum EvmError {
```
Comment (Elaixzar) [inline src/lib/chains/evm/evmErrors.ts:1]: Please do not use all caps in string fields.  Makes it seem like we are yelling at the developers.  All caps is ok for variable constant names, not in strings,

File: src/lib/chains/algorand/algoError.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+export enum AlgoError {
+    CLIENT_NOT_SET = 'CLIENT HAS NOT SET',
+    BRIDGE_NOT_SET = 'BRIDGE HAS NOT SET',
```
Comment (Elaixzar) [inline src/lib/chains/algorand/algoError.ts:3]: I like the centralization of error, but please do not use all caps in string fields. Makes it seem like we are yelling at the developers. All caps is ok for variable constant names, not in strings,

File: src/lib/chains/solana/solanaError.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+export enum SolanaError {
+    CLIENT_NOT_SET = 'CLIENT HAS NOT SET',
```
Comment (Elaixzar) [inline src/lib/chains/solana/solanaError.ts:2]: Same issue here, please do not use all capitals in error string.

Comment (Elaixzar) [reply src/lib/chains/solana/solanaError.ts:2]: Also, please check grammer:  

Client has not been set (or: Client not set)
Bridge has not been set (or Bridge not set)

File: src/lib/common/utils/utils.ts [hunk 4]
```diff
@@ -40,4 +44,15 @@ export const base64To0xString = (encoded: any) => {
...
 };
-
+export const  convertToAscii= (str:string) => {
+    let arg = Buffer.from(str, "base64").toString("ascii");
+    return arg;
+}
+ export const convertToNumber = (str:any) =>{
+    if (typeof str !== "number") {
+      str = Buffer.from(str, "base64");
+      return Number(algosdk.decodeUint64(str,"safe"));
```
Comment (Elaixzar) [inline src/lib/common/utils/utils.ts:54]: Shouldn't this be in Algorand/utils?  (since it specifically uses algosdk?)

File: tests/pollerTest.ts [hunk 5]
```diff
@@ -0,0 +1,248 @@
+// import {  BridgeTokens, Sleep } from "../src/lib/common";
```
Comment (Elaixzar) [inline tests/pollerTest.ts:1]: If this is no longer needed here, please remove.

**Review by Elaixzar — CHANGES_REQUESTED** [review]
Mostly good.  Please review the comments below.

**Comment by andropixels** [comment]
@saadjhk can you once review this branch's evm/connect.ts i am getting error, i might have missed something while merging !!

image
image

File: src/lib/chains/algorand/algoError.ts (hunk 2 above)
Comment (saadjhk) [reply src/lib/chains/algorand/algoError.ts]: @andropixels do not add whitespace within variable value, use `_`

File: src/lib/chains/evm/evmErrors.ts (hunk 1 above)
Comment (saadjhk) [reply src/lib/chains/evm/evmErrors.ts:1]: @Elaixzar all caps are indication of global constants

**Comment by saadjhk** [comment]
> @saadjhk can you once review this branch's evm/connect.ts i am getting error, i might have missed something while merging your changes !!
> 
> image image

`token.address` can be `undefined` so you need to make sure it is a string

File: src/lib/common/utils/value_units.ts [hunk 6]
```diff
@@ -27,6 +27,8 @@ export class ValueUnits {
       return returnValue;
     }
   }
+
```
Comment (saadjhk) [inline src/lib/common/utils/value_units.ts:30]: remove extra whitespaces/new lines

File: src/lib/chains/algorand/utils.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import algosdk from "algosdk";
+
+export const  convertToAscii= (str:string) => {
```
Comment (saadjhk) [inline src/lib/chains/algorand/utils.ts:3]: it'd be great if you add tests for these methods

File: src/lib/chains/algorand/algoError.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    UNDEFINED_ACCOUNTS ='Accounts not defined',
+    TIMEOUT ='Timeout- waiting for balance',
+    POLLER_NOT_SET='Algorand poller has not been set',
+    INVALID_MNEMONIC ='Invalid mnemonic',
+    ACCOUNT_INFO ='Account info not found',
+    INVALID_ACCOUNT ='Account is undefined',
+    INDEXER_NOT_SET ='Algo indexer has not been set',
+    INVALID_APP_ID ='App id not defined',
+    UNDEFINED_ROUTING ='Routing has not been set'
+
```
Comment (saadjhk) [inline src/lib/chains/algorand/algoError.ts:19]: remove extra newline

File: src/global.d.ts [hunk 9]
```diff
@@ -0,0 +1,18 @@
+
+export {};
```
Comment (saadjhk) [inline src/global.d.ts:2]: what is the use of this export?

**Review by saadjhk — CHANGES_REQUESTED** [review]

**Comment by andropixels** [comment]
> > @saadjhk can you once review this branch's evm/connect.ts i am getting error, i might have missed something while merging your changes !!
> > image image
> 
> `token.address` can be `undefined` so you need to make sure it is a string

strange on dev branch the undefined check is not added still there is no error !
please check evm/connect once in the dev branch as well

File: src/lib/chains/algorand/utils.ts (hunk 7 above)
Comment (andropixels) [reply src/lib/chains/algorand/utils.ts:3]: good suggetion will add

**Comment by saadjhk** [comment]
> > > @saadjhk can you once review this branch's evm/connect.ts i am getting error, i might have missed something while merging your changes !!
> > > image image
> > 
> > 
> > `token.address` can be `undefined` so you need to make sure it is a string
> 
> strange on dev branch the undefined check is not added still there is no error ! please check evm/connect once in the dev branch as well

yeah it needs to be `_token` as well, its just `token` here

File: src/global.d.ts (hunk 9 above)
Comment (andropixels) [reply src/global.d.ts:2]: This is required because TypeScript only allows augmentations to the global scope in module files.

Comment (saadjhk) [reply src/global.d.ts:2]: @andropixels there must be an issue in `tsconfig`, if you include a `.d.ts` file in your tsconfig you dont have to add an export in a definition file

Comment (andropixels) [reply src/global.d.ts:2]: image
its there am i missing anything ?

Comment (saadjhk) [reply src/global.d.ts:2]: look at `https://www.typescriptlang.org/tsconfig#typeRoots`

Comment (andropixels) [reply src/global.d.ts]: added a fix

**Pull request merged**